src/lib/opencv.ts

```diff
@@ -0,0 +1,174 @@
+import {RemoteLib} from './remote_lib';
+import Docker from 'dockerode';
+import fs, {createWriteStream} from 'fs';
+import {randomUUID} from 'crypto';
+import got from 'got';
+import stream from 'stream';
+import path from 'path';
+
+const OPENCV_IMAGE = 'hdgigante/python-opencv:4.9.0-alpine';
+const CACHE_FOLDER = 'cache';
+const ALLOWED_CALLS = ['imread', 'cvtColor', 'flip', 'rotate', 'blur', 'resize', 'Canny', 'imwrite'];
+
+export class ImageCache {
+  generateIdentifier(): string {
+    return randomUUID() + '.jpg';
+  }
+
+  getCachePath(key: string): string {
+    return `${CACHE_FOLDER}/${key}`;
+  }
+}
+
+interface FileArg {
+  fileType: string,
+  fileUrl: string,
+}
+
+function isFileArg(arg: any): arg is FileArg {
+  return 'object' === typeof arg && null !== arg && 'fileType' in arg;
+}
+
+const imageCache = new ImageCache();
+
+class EchoStream extends stream.Writable {
+  private content = '';
+  _write(chunk: any, encoding: BufferEncoding, callback: (error?: Error | null) => void): void {
+    // eslint-disable-next-line
+    this.content += chunk.toString();
+    callback();
+  }
+
+  getContent(): string {
+    return this.content;
+  }
+}
+
+export class OpenCvLib extends RemoteLib {
+  private dockerInstance: Docker|null = null;
+
+  async executeRemoteCall(callName: string, args: any[]): Promise<any> {
```

Comment:
Strange for me in typescript to work with `any` for args, that can only lead to behavior you have forgotten to support IMO. 
And of course also for return type, in normal TS code, there should be no `any` at all. (possibly `unknown` when the type is unknown as unknown force proper type assertion to cast back to a type)

Reply:
I've changed them to `unknown`, thanks :+1: